packages/gateway/src/features/search/type-resolvers.ts

```diff
@@ -195,12 +195,9 @@ const getGroomName = (source: ISearchDataTemplate) => {
 export const searchTypeResolvers: GQLResolver = {
   EventSearchSet: {
     __resolveType(obj: ISearchEventDataTemplate) {
-      if (obj._type === 'compositions' && obj._source.event === 'Birth') {
+      if (obj._source.event === 'Birth') {
```

Comment:
@Nil20 could you check if now the `_type` field in the object root isn't used anymore, the payload from the `search` service could only be whatever is inside that `_source` and nothing more? It's a bit of an abstraction leak that the `_type` & `_source` fields that are very Elasticsearch specific are referred in gateway

Reply:
[Here](https://github.com/opencrvs/opencrvs-core/blob/develop/packages/gateway/src/features/search/type-resolvers.ts#L213) in the type resolvers, also [here](https://github.com/opencrvs/opencrvs-core/blob/develop/packages/gateway/src/features/search/type-resolvers.ts#L273), also found for marriage. These ids are nothing but the `compositionId` of a record which is also available in the `_source` object. Should we change it here to use the `_source.compositionId` as you mentioned to prevent the abstraction leak?